### PR TITLE
Add support for using some other Site24x7 data center instead of the default US one

### PR DESCRIPTION
**Pull request by mpeltonen**
Not sure how these config options would be best introduced in README and/or examples, but I added something anyway. What do you think?

### CODE REVIEW

File: site24x7/provider.go [hunk 1]
```diff
@@ -50,6 +50,18 @@ func Provider() terraform.ResourceProvider {
 				Default:     4,
 				Description: "Maximum number of retries for Site24x7 API errors until giving up",
 			},
+			"api_base_url": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "https://www.site24x7.com/api",
```
Comment (martinohmann) [inline site24x7/provider.go:56]: Don't specify a default here, let `site24x7-go` do the defaulting when the `APIBaseURL` config is empty.

File: site24x7/provider.go [hunk 2]
```diff
@@ -50,6 +50,18 @@ func Provider() terraform.ResourceProvider {
...
+				Description: "Site24x7 API base url to use.",
+			},
+			"token_url": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "https://accounts.zoho.com/oauth/v2/token",
```
Comment (martinohmann) [inline site24x7/provider.go:62]: Don't specify a default here, let `site24x7-go` do the defaulting when the `TokenURL` config is empty.

File: _examples/main.tf [hunk 3]
```diff
@@ -22,6 +22,15 @@ provider "site24x7" {
...
   max_retries = 4
+
+  // Site24x7 API base URL to use. Used to select the desired data center.
+  // See https://www.site24x7.com/help/api/#introduction
+  api_base_url = "https://www.site24x7.com/api"
+
+  // Site24x7 token URL to use. Used to select the desired data center.
+  // See https://www.site24x7.com/help/api/#authentication
+  // NOTE: This needs to be configured to match the API base URL domain.
+  token_url = "https://accounts.zoho.com/oauth/v2/token"
```
Comment (martinohmann) [inline _examples/main.tf:33]: Please also update the example in `docs/index.md` with this information.

Also update the list of optional variables in this file to the following please:
```
- **api_base_url** (String) Site24x7 API base url to use.
- **max_retries** (Number) Maximum number of retries for Site24x7 API errors until giving up
- **retry_max_wait** (Number) Maximum wait time in seconds before retrying failed API requests (exponential backoff).
- **retry_min_wait** (Number) Minimum wait time in seconds before retrying failed API requests.
- **token_url** (String) Site24x7 OAuth token url to use.
```

**Review by martinohmann — CHANGES_REQUESTED** [review]
Thanks for the PR @mpeltonen! Only some minor comments.

File: site24x7/provider.go (hunk 1 above)
Comment (mpeltonen) [reply site24x7/provider.go]: Default removed.

File: site24x7/provider.go (hunk 2 above)
Comment (mpeltonen) [reply site24x7/provider.go]: Default removed.

File: _examples/main.tf (hunk 3 above)
Comment (mpeltonen) [reply _examples/main.tf:33]: Ah, didn't notice that file. Updated!

**Comment by martinohmann** [comment]
@mpeltonen thanks for the contribution! Will publish a new release shortly.

**Comment by martinohmann** [comment]
Version `0.0.5` was just published to the terraform registry: https://registry.terraform.io/providers/Bonial-International-GmbH/site24x7/0.0.5 @mpeltonen

**Comment by mpeltonen** [comment]
@martinohmann Uuh, initial registry release as a bonus! Thanks again, and thanks also for coding the provider in the first place 👍

**Pull request merged**